dspace-server-webapp/src/main/java/org/dspace/app/rest/authorization/impl/CanRegisterDOIFeature.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.authorization.impl;
+import java.sql.SQLException;
+
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureDocumentation;
+import org.dspace.app.rest.authorization.AuthorizeServiceRestUtil;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.security.DSpaceRestPermission;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Can the current user register a DOI for this item?
+ *
+ * @author Kim Shepherd
+ */
+@Component
+@AuthorizationFeatureDocumentation(name = CanRegisterDOIFeature.NAME,
+    description = "It can be used to verify if the user can register a DOI for this item")
+public class CanRegisterDOIFeature implements AuthorizationFeature {
+
+    @Autowired
+    private AuthorizeServiceRestUtil authorizeServiceRestUtil;
+
+    public static final String NAME = "canRegisterDOI";
```

Comment:
True. I'll address the DOI vs other identifiers as I can see this is a bit confusing and is an issue in other places you've picked up.

The endpoint is /identiifers, and the idea is a GET will retrieve all the identifiers, and a POST can act as a 'create' - which in the context of our pre-minted DOIs means moving them into "to be registered" status so that on the next DOIOrganiser run, they'll be registered with the provider.

You've noted below that these POSTs are empty, and this is true. I think I should update this to include an identifier type, even if only DOIs are supported right now, to make usage more clear and to help with future support for eg Ark or something.

In this case, the "canRegisterDOI" feature is tied specifically to a button that says "Register DOI". I think we could probably update the name of this feature for the purposes of authZ as it's likely an admin should be able to register any identifier type? But until we start to support other identifier types, I think the button should probably continue to say "Register DOI" just to make it really obvious what happens when you click it (maybe in future it can have a drop down with different identiier types that are valid for manual registration?).